### PR TITLE
【Prim】Custom softmax grad

### PR DESCRIPTION
**Pull request by JiabinYang**
### PR types

Others
### PR changes

Others
### Describe

Pcard-66975
This PR support softmax custom vjp.

### CODE REVIEW

File: paddle/phi/api/yaml/legacy_backward.yaml [hunk 1]
```diff
@@ -1144,6 +1144,7 @@
     param : [out]
   kernel :
     func : softmax_grad
+  composite : softmax_grad(out, out_grad, axis)
```
Comment (xiaoguoguo626807) [inline paddle/phi/api/yaml/legacy_backward.yaml]: add output

File: python/paddle/fluid/tests/unittests/prim/composite_ops/test_composite_softmax_custom_vjp.py [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+            tuple(
+                op.type
+                for op in paddle.jit.to_static(self.f)
+                .get_concrete_program()[1]
+                ._train_program.block(0)
+                .ops
+            ),
+        )
+        core._set_prim_all_enabled(False)
+
```
Comment (xiaoguoguo626807) [inline python/paddle/fluid/tests/unittests/prim/composite_ops/test_composite_softmax_custom_vjp.py:197]: no precision test?

File: paddle/phi/api/yaml/legacy_backward.yaml (hunk 1 above)
Comment (JiabinYang) [reply paddle/phi/api/yaml/legacy_backward.yaml]: done

File: python/paddle/fluid/tests/unittests/prim/composite_ops/test_composite_softmax_custom_vjp.py (hunk 2 above)
Comment (JiabinYang) [reply python/paddle/fluid/tests/unittests/prim/composite_ops/test_composite_softmax_custom_vjp.py:197]: done

**Review by xiaoguoguo626807 — APPROVED** [review]
LGTM

**Pull request merged**